### PR TITLE
dolphin-emu-beta: fix darwin build

### PR DESCRIPTION
**Pull request by IvarWithoutBones**
###### Description of changes

This fixes `dolphin-emu-beta` on darwin, everything seems to work in my tests. 

Things done:
* Ran `nixpkgs-fmt` on the derivation, the inputs being more spaced out will make it easier to read future diffs.
* Disabled the auto-updater, which requires XCode to compile. We'd want our version to be fixed anyways.
* Disabled generating a standalone executable instead of a regular application bundle, this would mean we cannot devendor libraries and because of that increase closure size by a massive amount.
* Enabled `wrapQtAppsHook` on darwin, without it icons do not show up.
* Added some new darwin-only dependencies that seem to be required now. `fmt_8` was added for both platforms, the version from the submodules fails to compile on darwin and i figured devendoring would be nice on linux too.
* Added myself as a maintainer.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [X] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and “core” functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by OPNA2608** [comment]
Errors for me on x86_64-darwin:

```
-- Checking for module 'minizip-ng>=3.0.0'
--   No package 'minizip-ng' found
-- Shared minizip not found, falling back to the static library
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for inttypes.h
-- Looking for inttypes.h - found
CMake Error at Externals/minizip/CMakeLists.txt:63 (check_function_exists):
  Unknown CMake command "check_function_exists".


-- Configuring incomplete, errors occurred!
```

Full log: [pr190373_dolphin-emu-beta.log](https://github.com/NixOS/nixpkgs/files/9531045/pr190373_dolphin-emu-beta.log)

**Comment by IvarWithoutBones** [comment]
Huh, that's weird. `nixpkgs-review` on my `x86_64-darwin` machine seems to finish cmake's setup phase just fine. Not sure what could be causing that, guess I'll devendor `minizip` to see if that fixes it.

**Comment by IvarWithoutBones** [comment]
@OPNA2608 I've added a patch that should fix this based on a thread from wireshark (https://seclists.org/wireshark/2021/Aug/7). Looks like `CheckFunctionExists` is needed for `check_function_exists`, which is included in the [`CMakeLists.txt.original`](https://github.com/dolphin-emu/dolphin/blob/c9896e1c4b95b8d8164e505037f3e887d9caf2fb/Externals/minizip/CMakeLists.txt.original#L76) file but not in the `CMakeLists.txt` that's actually used. 

Kind of annoying that I cannot reproduce this, I wonder what could be causing the impurity here. Only thing I could think of is Apple SDK frameworks, but those really should not influence CMake like this. Anyways, could you retry running `nixpkgs-review`?

File: pkgs/applications/emulators/dolphin-emu/master.nix [hunk 1]
```diff
@@ -1,12 +1,53 @@
...
+, enet
+, mbedtls
+, soundtouch
+, sfml
+, xz
+, writeScript
+, hidapi
+, fmt_8
+, vulkan-loader ? null
+, libpulseaudio ? null
```
Comment (SuperSandro2000) [inline pkgs/applications/emulators/dolphin-emu/master.nix:40]:
```suggestion
, vulkan-loader
, libpulseaudio
```

**Comment by IvarWithoutBones** [comment]
> Adding bzip2 to the buildInputs makes it build for me.

Done 👍

**Review by SuperSandro2000 — APPROVED** [review]

**Review by OPNA2608 — COMMENTED** [review]
Result of `nixpkgs-review pr 190373` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>dolphin-emu-beta</li>
  </ul>
</details>

---

`dolphin-emu-nogui --version` works and says `Dolphin 5.0-16793`, but I don't have any backups of my games on this machine to test and `Dolphin` complains about wanting at least macOS 10.14.0 (I'm on 10.13.6).

**Comment by SuperSamus** [comment]
Could the changes from #180567 be incorporated here?
I can't test Darwin, so I don't know how it's affected by it.
Maybe this could be also done to PrimeHack?

**Comment by IvarWithoutBones** [comment]
> dolphin-emu-nogui --version works and says Dolphin 5.0-16793, but I don't have any backups of my games on this machine to test and Dolphin complains about wanting at least macOS 10.14.0 (I'm on 10.13.6).

Interesting, I didn't know it only worked on newer MacOS versions. That `--version` flag seems useful for `testVersion`, I'll add that later 👍


> Could the changes from #180567 be incorporated here?
> I can't test Darwin, so I don't know how it's affected by it.



Sounds good, I'll add those changes here. I was planning on cleaning up the derivation and updating it in a future PR, so you having done the cleanup part already saves me some time :)

> Maybe this could be also done to PrimeHack?

That definitely seems useful, although id prefer for this PRs scope to stay at just `dolphin-emu-beta`. The build process for all these branches seems fairly similar anyways, I think the best solution is having a generic derivation we pass `src` and such to for all branches. Then fixes like these automatically apply to all of them.

That'll be a bit bigger of a change though, so I think its better to do it in a followup PR.

**Comment by IvarWithoutBones** [comment]
Things done:

* Added `passthru.tests.version`
* Applied https://github.com/NixOS/nixpkgs/pull/180567/commits/9c3dc20a5d65e3716b4b1ae9c49a0ba1244e5376 from @SuperSamus for some general cleanup

File: pkgs/applications/emulators/dolphin-emu/master.nix [hunk 2]
```diff
@@ -51,37 +134,51 @@ stdenv.mkDerivation rec {
...
+      update-source-version dolphin-emu-beta "$version" --rev="$rev"
+    '';
+  };
 
   meta = with lib; {
     homepage = "https://dolphin-emu.org";
     description = "Gamecube/Wii/Triforce emulator for x86_64 and ARMv8";
-    license = licenses.gpl2Plus;
-    maintainers = with maintainers; [ MP2E ashkitten xfix ];
+    mainProgram = "Dolphin";
```
Comment (SuperSamus) [inline pkgs/applications/emulators/dolphin-emu/master.nix:173]: I just noticed this, but isn't the `mainProgram` supposed to be `dolphin-emu`? Or is it different on Darwin?

**Pull request merged**